include/aspect/material_model/visco_plastic.h

```diff
@@ -261,6 +267,16 @@ namespace aspect
          */
         MaterialUtilities::PhaseFunction<dim> phase_function;
 
+        /**
+         * Determines whether to look up the dominant phases for each composition in its respective lookup table.
+         */
+        bool use_dominant_phase_for_viscosity;
+
+        /**
+         * Object that handles phase transitions.
+         */
+        MaterialUtilities::PhaseFunctionDiscrete<dim> phase_function_discrete;
```

Comment:
make this a `std::unique_ptr<MaterialUtilities::PhaseFunctionDiscrete<dim>>` like is done for the `rheology` variable above. Then only create the object if `use_dominant_phase_for_viscosity` is set to true in `parse_parameters` (exactly as for `rheology`). In the current implementation you will always create an object of type `MaterialUtilities::PhaseFunctionDiscrete<dim>`, but if `use_dominant_phase_for_viscosity` is false, then the object will not be initialized and will not parse any input parameters (in other words you carry an uninitialized object in a more or less undefined state around). This is not a problem as long as no one uses this object if `use_dominant_phase_for_viscosity`is false, but this is easy to forget. With my suggested change you will get an obvious error message if someone dereferences the unique pointer, while with your current implementation the object will do something (but we dont know what).

Reply:
Yes, that's a great suggestion. Thanks for the explanation.